infra/concourse/build/developer-tools/build/scripts/task_helper_functions.sh

```diff
@@ -177,6 +177,44 @@ check_whitespace() {
   return $((rc+$?))
 }
 
+# Helper function to facilitate switch to a 0.12 doc generator
+# which has `different` markers and `different` helper scripts
```

Comment:
don't understand comment what 'different' means here? different from what?

Reply:
Oh, I didn't explain it further as it looks obvious form the code:
- markers replaced with `sed` [here](https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/285/files/7c94a6e5b5089bdbec6e3ed43a9d34394aa5e585#diff-f66a9091f6920e308f374bf1dfb9609dR189-R190)
- old generator helper [here](https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/285/files/7c94a6e5b5089bdbec6e3ed43a9d34394aa5e585#diff-f66a9091f6920e308f374bf1dfb9609dR194) replaced with 1 new helper script and 1 additional validator [here](https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/285/files/7c94a6e5b5089bdbec6e3ed43a9d34394aa5e585#diff-f66a9091f6920e308f374bf1dfb9609dR198)

More than that: each block of the code is commented
- [replace markers](https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/285/files/7c94a6e5b5089bdbec6e3ed43a9d34394aa5e585#diff-f66a9091f6920e308f374bf1dfb9609dR185)
- [replace scripts](https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/285/files/7c94a6e5b5089bdbec6e3ed43a9d34394aa5e585#diff-f66a9091f6920e308f374bf1dfb9609dR193) 

I think giving even more details in initial comment is a little too much. But I'm open for any changes.

Reply:
well, my question means, does this comment will be relevant in future? in one month for example, new developer will not understand what is difference from what by this comment

Reply:
> well, my question means, does this comment will be relevant in future? in one month for example, new developer will not understand what is difference from what by this comment

`bash` is probably the easiest language to understand :) 
Obviously function named `replace_doc_generator` must be replacing something. 
What exactly and how is pretty much self-descriptive. But I did add comments for those few lines which replace particular things.

How can I be more specific in the comment? I don't mind changing it.

In one month we will probably have to reverse the function or create another one which does exactly opposite switch. As we're using pre-commit terraform workaround just because terrafrom_doc doesn't support 0.12 syntax..

Reply:
@ivankorn  I did not speak about function name

>  which has `different` markers and `different` helper scripts

what will 'different' means for developer that will start working in 2 month from now?
don't want to argue on that, approve

Reply:
> what will 'different' means for developer that will start working in 2 month from now?
> don't want to argue on that, approve
@kopachevsky I tried to extend the description [here](https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/compare/7c94a6e5b5089bdbec6e3ed43a9d34394aa5e585..5e73e00cb1033b9ccc2c85aa86171ed483d82c56)